analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -165,7 +165,7 @@ private static ProgramState ProcessAssertedBoolSymbol(ProgramState state, IOpera
         else
         {
             return operation.TrackedSymbol() is { } symbol
-                ? state.SetSymbolConstraint(symbol, BoolConstraint.From(!isNegated))
+                ? state.SetSymbolConstraint(symbol, BoolConstraint.From(!isNegated)).SetSymbolConstraint(symbol, ObjectConstraint.NotNull)
```

Comment:
I can not see `[DataRow("arg", "bool")]` or `[DataRow("!arg", "bool")]` tests. Do we have these? If so, they should have changed with this change.

Reply:
I've rewrote the assertions to make this explicit